authnz/src/main/java/io/stargate/auth/UnauthorizedException.java

```diff
@@ -15,7 +15,7 @@
  */
 package io.stargate.auth;
 
-public class UnauthorizedException extends Exception {
+public class UnauthorizedException extends RuntimeException {
```

Comment:
This is way more convenient for the async code.

The downside is that clients aren't forced to handle the error explicitly anymore. But from a quick scan of the code, it looks like in most places this was handled in a generic way anyway (`catch Exception`, Jersey exception handler, etc).